app/models/notification.rb

```diff
@@ -136,6 +136,20 @@ def update_notifications(notifiable, action = nil)
       Notifications::UpdateJob.perform_later(notifiable.id, notifiable.class.name, action)
     end
 
+    def fast_destroy_old_notifications(destroy_before_timestamp = 4.months.ago)
+      notification_sql = <<-SQL
+        DELETE FROM notifications
+        WHERE notifications.id IN (
+          SELECT notifications.id
+          FROM notifications
+          WHERE created_at < '#{destroy_before_timestamp}'
+          LIMIT 10000
```

Comment:
I arbitrarily picked 10k here as a good starting point and figured we could tweak it if we need to. Open to other suggestions though!